src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImpl.java

```diff
@@ -73,12 +158,35 @@ public void executeScript(
     }
   }
 
+  @VisibleForTesting
+  static String getUtcTimeStringFromTimestamp(Timestamp timestamp) {
+    Instant instant = timestamp.toInstant();
+    // Remove the "Z" (that signifies UTC timezone) and the separators that are not necessary for
+    // human interpretation.
+    String instantWithoutNanoseconds =
+        instant.truncatedTo(ChronoUnit.SECONDS).toString().replaceAll("[Z\\-:]", "").trim();
+    // Keep 6 digits of the fractional seconds from nano, which is equivalent to round(nanos / 10^9
+    // * 10^6) with left-padded zeros. Prepend with S to separate from seconds.
+    String sixDigitNanos =
+        String.format((Locale) null, "S%06d", Math.round(timestamp.getNanos() / Math.pow(10, 3)));
+    return instantWithoutNanoseconds + sixDigitNanos;
+  }
+
   @Override
-  public String getScript(SqlTemplateRenderer sqlTemplateRenderer, String scriptName) {
+  public String getScript(
+      SqlTemplateRenderer sqlTemplateRenderer,
+      String scriptName,
+      ImmutableList<String> sortingColumns) {
     Preconditions.checkArgument(
         scriptsMap.containsKey(scriptName),
         String.format("Script name %s is not available.", scriptName));
-    return sqlTemplateRenderer.renderTemplate(scriptName, scriptsMap.get(scriptName).get());
+    if (!sortingColumns.isEmpty()) {
+      sqlTemplateRenderer.getSqlScriptVariablesBuilder().setSortingColumns(sortingColumns);
+    }
+    String rendered =
+        sqlTemplateRenderer.renderTemplate(scriptName, scriptsMap.get(scriptName).get());
+    sqlTemplateRenderer.getSqlScriptVariablesBuilder().setSortingColumns(ImmutableList.of());
```

Comment:
Did you consider sorting columns injection in the `getSqlTemplateRenderer` method?

Reply:
I decided not to inject the sorting columns during getSqlTemplateRenderer based on these considerations:
1. sortingColumns serves the chunked-writing mechanism only and is hard-coded by design, i.e. should not be modifiable by the users, and thus should be separated from Vars of SqlScriptVariables.
2. For the same reason, sortingColumns should be tied to the ScriptLoader/ScriptManager instead of in ExtractExecutor.

It is true that resetting it to empty was not needed during executeScript because a new sqlTemplateRenderer is supplied every time; fixed that part.